websites/N/namu.wiki/presence.ts

```diff
@@ -0,0 +1,181 @@
+  "backlink": "View Backlink",
+  "contribution": "View Contribution",
+  "Policy": "View Policy",
+  "Search": "Searching",
+
+  // edit
+  "edit": "Editing",
+  "move": "Move Document",
+  "delete": "Delete Confirm",
+  "new_edit_request": "Create New Edit Request",
+
+  // discuss
+  "discuss": "Dicsussing",
+  "thread": "View Discuss",
+  "edit_request": "Edit Request",
+
+  // recent
+  "RecentDiscuss": "View Recent Discuss",
+  "RecentChanges": "View Recent Changes",
+
+  // not used well
+  "NeededPages": "View Needed Pages",
+  "OrphanedPages": "View Orphaned Pages",
+  "UncategorizedPages": "View Uncategorized Pages",
+  "OldPages": "View Old Pages",
+  "ShortestPages": "View Shortest Pages",
+  "LongestPages": "View Longest Pages",
+  "BlockHistory": "View Block History",
+  "RandomPage": "View Random Page",
+  "Upload": "Upload file",
+  "License": "View License"
+},
+// /member/(action)
+validateMembershipUrl = /\/member\/(.+)/,
+membersMapping: interfaceMapping = {
+  "login": "Login",
+  "recover_password": "Recover Password",
+  "signup": "Sign Up",
+  "mypage": "My Page",
+  "change_email": "Change Email",
+  "change_password": "Change Password",
+  "activate_otp": "Activate OTP"
+},
+// /RecentChanges?logtype=(search)
+changesMapping: interfaceMapping = {
+  "all": "All Changes",
+  "create": "Created Documents",
+  "delete": "Deleted Documents",
+  "move": "Moved Documents",
+  "revert": "Reverted Documents"
+},
+// /RecentDiscuss?logtype=(search)
+discussMapping: interfaceMapping = {
+  "normal_thread": "Normal Thread",
+  "old_thread": "Old Thread",
+  "closed_thread": "Closed Thread",
+  "open_editrequest": "Opened Edit Request",
+  "accepted_editrequest": "Accepted Edit Request",
+  "closed_editrequest": "Closed Edit Request",
+  "old_editrequest": "Old Edit Request"
+},
+// /contribution/(type)/(username)/(contributeType)
+validateContributeUrl = /\/contribution\/(.+)\/(.+)\/(.+)/;
+
+let currentPage = document.location.pathname, currentTime = Date.now();
+presence.on("UpdateData", async () => {
+  const privacy = await presence.getSetting("privacy"),
+  showTimestamp = await presence.getSetting("show_timestamp"),
+  presenceData: PresenceData = {
+    largeImageKey: "namu"
+  };
+
+  if (document.location.hostname !== "namu.wiki") return;
+  const path = document.location.pathname, parsedUrl = path.split("/"); // It's a very bad design, but they have a slash document.
+  if (parsedUrl === null) return;
+
+  /**
+   * 
+   * Setting Details & State
+   * 
+   */
+  let privateMode = false;
+  const action = parsedUrl[1], details = boardTypeMapping[action];
+  presenceData.details = ((details === undefined) ? "Unknown Action" : details);
+
+  let page;
+  /* View Contribute */
+  if (page = validateContributeUrl.exec(path)) {
+    // The IPv4 regex should not be used. it may be IPv6.
+    let isUser = (page[1] === "author");
+    page = `${isUser ? "User" : "IP"}: ${isUser ? page[2] : page[2].replace(/(?<=.{3})./gi, "*")}`;
```

Comment:
Why do you need to show the ip?

Reply:
namu.wiki has a page that displays user's IP.
Do not display user's IP. Mask the IP.

(My english is very poor :cry:)

Reply:
Can't you just remove the ip from however you are getting it?

Reply:
Ok so, you could just do a split statement where it removes the ip from the author names.

Reply:
e.g xxx.xxx.split("1")[0];

Reply:
Ah ok.

Reply:
Although, it is not necessary to show it in the presence. You can just remove it.

Reply:
ok i will change it

Reply:
Give it a go. Hopefully works. Anyways, the actual reviewers will review your code soon. They may or may not like you to display the IP of users. Good luck 😃